code/modules/antagonists/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,1095 @@
+				thrallToRevive.mind.RemoveSpell(/obj/effect/proc_holder/spell/targeted/lesser_glare)
+				thrallToRevive.mind.RemoveSpell(/obj/effect/proc_holder/spell/self/lesser_shadow_walk)
+				thrallToRevive.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/sling/glare(null))
+				thrallToRevive.mind.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/veil(null))
+			if("Revive")
+				if(!is_thrall(thrallToRevive))
+					to_chat(user, "<span class='warning'>[thrallToRevive] is not a thrall.</span>")
+					revert_cast()
+					return
+				if(thrallToRevive.stat != DEAD)
+					to_chat(user, "<span class='warning'>[thrallToRevive] is not dead.</span>")
+					revert_cast()
+					return
+				if(HAS_TRAIT(thrallToRevive, TRAIT_BADDNA))
+					to_chat(user, "<span class='warning'>[thrallToRevive] is too far gone.</span>")
+					revert_cast()
+					return
+				user.visible_message("<span class='danger'>[user] kneels over [thrallToRevive], placing their hands on \his chest.</span>", \
+									"<span class='shadowling'>You crouch over the body of your thrall and begin gathering energy...</span>")
+				thrallToRevive.notify_ghost_cloning("Your masters are resuscitating you! Re-enter your corpse if you wish to be brought to life.", source = thrallToRevive)
+				if(!do_mob(user, thrallToRevive, 30))
+					to_chat(user, "<span class='warning'>Your concentration snaps. The flow of energy ebbs.</span>")
+					revert_cast()
+					return
+				to_chat(user, "<span class='shadowling'><b><i>You release a massive surge of power into [thrallToRevive]!</b></i></span>")
+				user.visible_message("<span class='boldannounce'><i>Red lightning surges from [user]'s hands into [thrallToRevive]'s chest!</i></span>")
+				playsound(thrallToRevive, 'sound/weapons/Egloves.ogg', 50, 1)
+				playsound(thrallToRevive, 'sound/machines/defib_zap.ogg', 50, 1)
+				user.Beam(thrallToRevive,icon_state="red_lightning",time=1)
+				var/b = do_mob(user, thrallToRevive, 20)
+				if(thrallToRevive.revive(full_heal = 1))
+					thrallToRevive.visible_message("<span class='boldannounce'>[thrallToRevive] heaves in breath, dim red light shining in their eyes.</span>", \
+											   "<span class='shadowling'><b><i>You have returned. One of your masters has brought you from the darkness beyond.</b></i></span>")
+					thrallToRevive.Knockdown(4)
+					thrallToRevive.emote("gasp")
+					playsound(thrallToRevive, "bodyfall", 50, 1)
+					if (!b)
+						thrallToRevive.Knockdown(50)
+						thrallToRevive.Unconscious(500)
+						thrallToRevive.visible_message("<span class='boldannounce'>[thrallToRevive] collapses in exhaustion.</span>", \
+						   "<span class='warning'><b><i>You collapse in exhaustion... nap..... dark.</b></i></span>")
+			else
+				revert_cast()
+				return
+
+/obj/effect/proc_holder/spell/targeted/shadowling_extend_shuttle
+	name = "Destroy Engines"
+	desc = "Sacrifice a thrall to extend the time of the emergency shuttle's arrival by fifteen minutes. This can only be used once."
```

Comment:
Either it should be way less time, or it should deduct the time upon the owner's death

Reply:
Decreased to 10 minutes